lib/good_job/performer.rb

```diff
@@ -1,18 +1,46 @@
 module GoodJob
+  #
+  # Performer queries the database for jobs and performs them on behalf of a
```

Comment:
This is the applied purpose for the Perfomer. Though the performer is intended to be generic/ignorant of the nature of the task. 

The practical reason for the performer is to enforce a callable that does not rely on scoped/closures variables because they might not be available when executed in a different thread. Or alternatively, it describes an interface that is compatible with the Scheduler.